### PR TITLE
Fixed compile-time warnings for ESP32 and ESP32-S2

### PR DESCRIPTION
**Pull request by laukik-hase**
Description
-----------
Fixed the compile-time warnings which are observed when AFR is built for ESP32 and ESP32-S2. 

Checklist:
----------
- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/espressif/boards/esp32/CMakeLists.txt [hunk 1]
```diff
@@ -543,6 +543,54 @@ set_source_files_properties(${AFR_ROOT_DIR}/libraries/c_sdk/standard/mqtt/test/u
...
+    PROPERTIES COMPILE_FLAGS
+    "-Wno-array-bounds")
+
+set_source_files_properties(
+    ${AFR_ROOT_DIR}/libraries/ble/test/iot_test_ble_end_to_end.c
+    ${AFR_ROOT_DIR}/libraries/ble/test/iot_mqtt_ble_system_test.c
+    ${AFR_ROOT_DIR}/libraries/c_sdk/standard/common/iot_init.c
+    ${AFR_ROOT_DIR}/libraries/c_sdk/standard/common/iot_device_metrics.c
+    ${AFR_ROOT_DIR}/libraries/abstractions/secure_sockets/lwip/iot_secure_sockets.c
+    ${AFR_ROOT_DIR}/vendors/espressif/boards/ports/wifi/iot_softap_wifi_provisioning.c
```
Comment (shubhamkulkarni97) [inline vendors/espressif/boards/esp32/CMakeLists.txt:556]: This file is under espressif port, can we actually remove unused function?

File: vendors/espressif/boards/esp32/aws_demos/config_files/ota_demo_config.h [hunk 2]
```diff
@@ -71,6 +71,7 @@
  * bytes are read from network interface. Keeping this timeout to a sufficiently
  * large value so as to account for delay of receipt of a large block of message.
  */
+#undef MQTT_RECV_POLLING_TIMEOUT_MS
```
Comment (shubhamkulkarni97) [inline vendors/espressif/boards/esp32/aws_demos/config_files/ota_demo_config.h:74]:
```suggestion
#ifndef MQTT_RECV_POLLING_TIMEOUT_MS
#define MQTT_RECV_POLLING_TIMEOUT_MS ( 1000U )
#endif
```

Consider this suggestion for all occurrences

**Review by shubhamkulkarni97 — CHANGES_REQUESTED** [review]
@laukik-hase I have left minor comments, overall LGTM

**Comment by laukik-hase** [comment]
This PR suppresses the compiler warnings in Amazon FreeRTOS library when building for ESP32. However, rather than suppressing those warnings at compile time, the code itself needs to be fixed at each site. 

As per #1629, is this issue being still tracked internally?

**Comment by shubhamkulkarni97** [comment]
@aggarg @mingyue86010 Could you take a look at this PR? It fixes build warnings for Espressif platforms

**Review by aggarg — DISMISSED** [review]

**Comment by laukik-hase** [comment]
Updated style changes to fix CI failures

**Review by aggarg — APPROVED** [review]

**Pull request merged**